From ae60e0cd42844909b603d1e232d267055203dcfb Mon Sep 17 00:00:00 2001
From: Samreen Nilofer <samreen.nilofer@intel.com>
Date: Tue, 23 Jun 2015 16:26:38 +0530
Subject: [PATCH] ASoC: Intel: Add separate handler

- Add a helper function for setting clock
and tdm slot

Change-Id: If8bc05571d0b1da372d9d24f7b6ac503ee6fd5d6
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-12531
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 sound/soc/intel/board/cht_bl_dpcm_rt5672.c | 22 +++++++++++++++-------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
index 2468be8efbae..360d5e89f2de 100644
--- a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
+++ b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
@@ -523,11 +523,9 @@ static const struct snd_kcontrol_new cht_mc_controls[] = {
 };
 
 
-static int cht_aif1_hw_params(struct snd_pcm_substream *substream,
-			     struct snd_pcm_hw_params *params)
+static int cht_set_slot_fmt_clk(struct snd_soc_dai *codec_dai,
+				unsigned int rate)
 {
-	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_soc_dai *codec_dai = rtd->codec_dai;
 	unsigned int fmt;
 	int ret;
 
@@ -559,22 +557,32 @@ static int cht_aif1_hw_params(struct snd_pcm_substream *substream,
 	}
 
 	ret = snd_soc_dai_set_pll(codec_dai, 0, RT5670_PLL1_S_MCLK,
-				  CHT_PLAT_CLK_3_HZ, params_rate(params) * 512);
+				  CHT_PLAT_CLK_3_HZ, rate * 512);
 	if (ret < 0) {
 		pr_err("can't set codec pll: %d\n", ret);
 		return ret;
 	}
 
 	ret = snd_soc_dai_set_sysclk(codec_dai, RT5670_SCLK_S_PLL1,
-				     params_rate(params) * 512,
-				     SND_SOC_CLOCK_IN);
+				     rate * 512, SND_SOC_CLOCK_IN);
 	if (ret < 0) {
 		pr_err("can't set codec sysclk: %d\n", ret);
 		return ret;
 	}
+
 	return 0;
 }
 
+static int cht_aif1_hw_params(struct snd_pcm_substream *substream,
+			     struct snd_pcm_hw_params *params)
+{
+	struct snd_soc_pcm_runtime *rtd = substream->private_data;
+	struct snd_soc_dai *codec_dai = rtd->codec_dai;
+
+	pr_debug("Enter:%s", __func__);
+	return cht_set_slot_fmt_clk(codec_dai, params_rate(params));
+}
+
 static int cht_compr_set_params(struct snd_compr_stream *cstream)
 {
 	return 0;
-- 
1.9.1

